src/scenario.rs

```diff
@@ -8,6 +8,8 @@ use crate::names::*;
 mod no_extra;
 use no_extra::NoExtra;
 
+pub(crate) mod visualization;
```

Comment:
is there anything in a `Scenario` that is not `pub` enough, so that it is a submodule here?

I would put it to `crate::visualization`.